### ISSUE TITLE
difference between bayestestR::hdi() and sjstats::hdi()

### ISSUE THREAD

**Issue by strengejacke** [easystats/bayestestR#39]
I found a strange inconsistency for `hdi()` with brms-models, but I'm not sure if this is only happening for brms-models or not? I'm not sure in how far your implementation of HDI differs from mine (which is adapted from Kruscke 2015).

stanreg = identical
brms = differences

``` r
m1 <- insight::download_model("stanreg_lm_1")

as.data.frame(bayestestR::hdi(m1, ci = .95))
#>     Parameter CI    CI_low   CI_high
#> 1 (Intercept) 95 33.641431 41.069859
#> 2          wt 95 -6.418331 -4.236012

as.data.frame(sjstats::hdi(m1, prob = .95))
#>          term   hdi.low  hdi.high
#> 1 (Intercept) 33.641431 41.069859
#> 2          wt -6.418331 -4.236012

set.seed(123)
library(brms)
b4 <- brms::brm(mpg ~ wt + cyl + gear, data = mtcars, iter = 500, chains = 1)

as.data.frame(bayestestR::hdi(b4, ci = .95))
#>     Parameter CI    CI_low    CI_high
#> 1 b_Intercept 95 34.996964 50.9281495
#> 2        b_wt 95 -5.339557 -1.9348616
#> 3       b_cyl 95 -2.380895 -0.6869611
#> 4      b_gear 95 -1.790944  1.0685669

as.data.frame(sjstats::hdi(b4, prob = .95))
#>          term   hdi.low   hdi.high
#> 1 b_Intercept 34.549999 50.8355624
#> 2        b_wt -5.373982 -1.9348616
#> 3       b_cyl -2.295808 -0.5923283
#> 4      b_gear -1.845044  1.0685669
```

<sup>Created on 2019-03-14 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

**Comment by DominiqueMakowski** [comment]
That's weird... I will look into it

**Comment by DominiqueMakowski** [comment]
The difference might come from the fact that sjstats uses 👍 
[`ci.index <- ceiling(prob * length(x))`](https://github.com/strengejacke/sjstats/blob/f5713a067f39e4656337d2e217b14551d090197d/R/HDI.R#L357) whereas in bayestestR we use `window_size <- floor(ci * length(x_sorted))`.

I am not sure which one is the most correct though...

From docs:
- ceiling takes a single numeric argument x and returns a numeric vector containing the smallest integers not less than the corresponding elements of x.
- floor takes a single numeric argument x and returns a numeric vector containing the largest integers not greater than the corresponding elements of x

Basically in one case for instance the value is `237.5` and one rounds it to 237 and the other one to 238...

**Comment by DominiqueMakowski** [comment]
If HDInterval they seemed to favour floor (?):

https://www.rdocumentation.org/packages/HDInterval/versions/0.2.0/source

**Comment by strengejacke** [comment]
Ok, Kruschke uses `ceiling`.

![image](https://user-images.githubusercontent.com/26301769/54415269-15071780-46fc-11e9-82a2-75a2d6a2f576.png)
![image](https://user-images.githubusercontent.com/26301769/54415281-1d5f5280-46fc-11e9-8c5b-0276e2716cbd.png)

**Comment by strengejacke** [comment]
I'm ok with both, if both are ok, but we should mention the difference in the docstrings.

**Comment by DominiqueMakowski** [comment]
Well in HDInterval they seem to have explicitly commented out the ceiling:

```
# exclude <- ceiling(n * (1 - credMass)) # Not always the same as...
exclude <- n - floor(n * credMass)       # Number of values to exclude
```
We could eventually ask them the rationale for it, out of curiosity

**Comment by DominiqueMakowski** [comment]
Here's a very comprehensive [clarification](https://github.com/mikemeredith/HDInterval/issues/4#event-2209153708) from the author of HDInterval. It seems that Kruschke's use of ceiling is more recent than its floor counterpart, and that it gives, if different, more conservative CIs. I think that it would be consistent with other functions (going for conservative solutions) to switch to `ceiling`.

**Comment by DominiqueMakowski** [comment]
edit from @mikemeredith, it seems that floor is more conservative. @strengejacke Any preference then?

**Comment by strengejacke** [comment]
Let's keep it as we have it now (using floor), or not? I have no strong preference, I would just mention it in the docs.

**Comment by DominiqueMakowski** [comment]
Or maybe stick with Kruschke's version, it will be easier to explain and reference
